include/aws/http/private/h2_decoder.h

```diff
@@ -58,9 +58,10 @@ struct aws_h2_decoder_vtable {
         void *userdata);
     struct aws_h2err (*on_push_promise_end)(uint32_t stream_id, bool malformed, void *userdata);
 
-    /* For DATA frame: _begin() is called, then 0+ _i() calls, then _end().
+    /* For DATA frame: _begin() is called, then 0 or 1 _padding_len() calls, then 0+ _i() calls, then _end().
      * No other decoder callbacks will occur in this time */
     struct aws_h2err (*on_data_begin)(uint32_t stream_id, uint32_t payload_len, bool end_stream, void *userdata);
+    struct aws_h2err (*on_data_padding_len)(uint32_t stream_id, uint8_t padding_len, void *userdata);
```

Comment:
ugh... Edge case: PADDED frame to have pad length of zero🤯 
I guess it's better to be payload_len instead of data_len as we will not tell user anyway.
`struct aws_h2err (*on_data_begin)(uint32_t stream_id, uint32_t payload_len, uint16_t padding_len, bool end_stream, void *userdata);`